accounts/abi/bind/backends/simulated.go

```diff
@@ -130,6 +134,65 @@ func (b *SimulatedBackend) rollback() {
 	b.pendingState, _ = state.New(b.pendingBlock.Root(), b.blockchain.StateCache(), nil)
 }
 
+// Reorg simulates a chain reorg with parameters depth and length.
+// f can be used to insert, reorder and exclude transactions.
+// f receives a slice of Transactions where each slice entry contains the
+// transactions that were contained in the orhpaned block.
+// The slice that is passed to f has length depth, and the one returned
+// must have at least length length.
+// Panics if length is not greater than depth.
+func (b *SimulatedBackend) Reorg(ctx context.Context, depth, length int64, f Reorderer) error {
```

Comment:
F could be inlined (doesn't have to be, but would fit better with the rest of geth).
Also please give f a more descriptive name (`reorder` maybe)